x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -31,61 +35,36 @@ private JwtRealmSettings() {}
 
     public static final String TYPE = "jwt";
 
-    /**
-     * HMAC length NOT by specific SHA-2 lengths.
-     * Example: final byte[] hmacKeyBytes = new byte[randomIntBetween(32,100)];
-     */
-    public static final List<String> SUPPORTED_SECRET_KEY_SIGNATURE_ALGORITHMS = List.of("HS256", "HS384", "HS512");
-
-    /**
-     * RSA lengths NOT by specific SHA-2 lengths.
-     * Example: final Integer rsaBits = randomFrom(2048, 3072, 4096, 6144);
-     */
-    public static final List<String> SUPPORTED_PUBLIC_KEY_RSA_SIGNATURE_ALGORITHMS = List.of(
-        "RS256",
-        "RS384",
-        "RS512",
-        "PS256",
-        "PS384",
-        "PS512"
-    );
-
-    /**
-     * EC curves by specific SHA-2 lengths.
-     * Example: final Curve ecCurve = randomFrom(Curve.forJWSAlgorithm(jwsAlgorithm));
-     * Note: Excludes ES256K. The library supports it, but Java 11.0.9, 11.0.10, 17 disabled support for it by default.
-     */
-    public static final List<String> SUPPORTED_PUBLIC_KEY_EC_SIGNATURE_ALGORITHMS = List.of("ES256", "ES384", "ES512");
-
-    public static final List<String> SUPPORTED_PUBLIC_KEY_SIGNATURE_ALGORITHMS = Stream.of(
-        SUPPORTED_PUBLIC_KEY_RSA_SIGNATURE_ALGORITHMS,
-        SUPPORTED_PUBLIC_KEY_EC_SIGNATURE_ALGORITHMS
+    // Signature algorithms
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS_HMAC = List.of("HS256", "HS384", "HS512");
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS_RSA = List.of("RS256", "RS384", "RS512", "PS256", "PS384", "PS512");
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS_EC = List.of("ES256", "ES384", "ES512");
+    public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS_PKC = Stream.of(
+        SUPPORTED_SIGNATURE_ALGORITHMS_RSA,
+        SUPPORTED_SIGNATURE_ALGORITHMS_EC
     ).flatMap(Collection::stream).toList();
-
     public static final List<String> SUPPORTED_SIGNATURE_ALGORITHMS = Stream.of(
-        SUPPORTED_SECRET_KEY_SIGNATURE_ALGORITHMS,
-        SUPPORTED_PUBLIC_KEY_SIGNATURE_ALGORITHMS
+        SUPPORTED_SIGNATURE_ALGORITHMS_HMAC,
+        SUPPORTED_SIGNATURE_ALGORITHMS_PKC
     ).flatMap(Collection::stream).toList();
 
     // Header names
     public static final String HEADER_END_USER_AUTHORIZATION = "Authorization";
-    public static final String HEADER_END_USER_AUTHORIZATION_SCHEME = "Bearer";
-
     public static final String HEADER_CLIENT_AUTHORIZATION = "X-Client-Authorization";
-
-    public static final String SUPPORTED_CLIENT_AUTHORIZATION_TYPE_SHARED_SECRET = "SharedSecret";
-    public static final String SUPPORTED_CLIENT_AUTHORIZATION_TYPE_NONE = "None";
-    public static final List<String> SUPPORTED_CLIENT_AUTHORIZATION_TYPES = List.of(
-        SUPPORTED_CLIENT_AUTHORIZATION_TYPE_SHARED_SECRET,
-        SUPPORTED_CLIENT_AUTHORIZATION_TYPE_NONE
+    public static final String HEADER_END_USER_AUTHORIZATION_SCHEME = "Bearer";
```

Comment:
Headers are client secret and JWT. I tried to use client vs end-user naming for these terms, respectively, so end-user refers to the JWT.

Reply:
Gotcha. I'd personally have this as a package protected member of the `JWTRealm` class along with `HEADER_END_USER_AUTHORIZATION` since this is where these are used , and only for the purpose of not hardcoding the string itself where needed. It's not a setting, it's not configurable, so I don't think there's a reason to keep this in JwtRealmSettings, WDYT ?

Reply:
These settings can be moved to JwtRealm.
```
    public static final String HEADER_END_USER_AUTHORIZATION = "Authorization";
    public static final String HEADER_CLIENT_AUTHORIZATION = "X-Client-Authorization";
    public static final String HEADER_END_USER_AUTHORIZATION_SCHEME = "Bearer";
```

Other settings cannot be moved to JwtRealm, because that would create a circular dependency between the two classes.

If that is sufficient, I can make the change. However, an alternative is move them all to a separate class, perhaps called JwtConstants or JwtRealmConstants. That would allow grouping them all in one place, with the risk of circular dependencies.

Reply:
I'm happy if we can move anything that is not settings related or just a constant. I don't think we need a new class, the Realm class should be just fine